test/e2e/set/scyllacluster/scyllacluster_expose.go

```diff
@@ -4,26 +4,258 @@ package scyllacluster
+
+		svc, err = f.KubeClient().CoreV1().Services(sc.Namespace).Get(ctx, svcName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e.validateService(svc)
+
+		pod, err := f.KubeClient().CoreV1().Pods(sc.Namespace).Get(ctx, svcName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		configClient, err := utils.GetScyllaConfigClient(ctx, f.KubeClient().CoreV1(), sc, pod.Status.PodIP)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		e.validateScyllaConfig(ctx, configClient, svc, pod)
+
+		if utilfeature.DefaultMutableFeatureGate.Enabled(features.AutomaticTLSCertificates) {
+			serviceAndPodIPs, err := utils.GetNodesServiceAndPodIPs(ctx, f.KubeClient().CoreV1(), sc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			hostsIPs, err := helpers.ParseClusterIPs(serviceAndPodIPs)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			var servingDNSNames []string
+			services, err := f.KubeClient().CoreV1().Services(sc.Namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(naming.ClusterLabels(sc)).String(),
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, svc := range services.Items {
+				servingDNSNames = append(servingDNSNames, fmt.Sprintf("%s.%s.svc", svc.Name, svc.Namespace))
+			}
+
+			servingCertSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-serving-certs", sc.Name), metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			servingCerts, _, _, _ := verification.VerifyAndParseTLSCert(servingCertSecret, verification.TLSCertOptions{
+				IsCA:     pointer.Ptr(false),
+				KeyUsage: pointer.Ptr(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+			})
+			o.Expect(servingCerts).To(o.HaveLen(1))
+			o.Expect(servingCerts[0].Subject.CommonName).To(o.BeEmpty())
+			o.Expect(helpers.NormalizeIPs(servingCerts[0].IPAddresses)).To(o.ConsistOf(hostsIPs))
+			o.Expect(servingCerts[0].DNSNames).To(o.ConsistOf(servingDNSNames))
+		}
+	},
+		g.Entry("using ClusterIP Service, and broadcast it to both client and nodes", &entry{
+			exposeOptions: &scyllav1.ExposeOptions{
+				NodeService: &scyllav1.NodeServiceTemplate{
+					Type: scyllav1.NodeServiceTypeClusterIP,
+				},
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypeServiceClusterIP,
+					},
+					Clients: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypeServiceClusterIP,
+					},
+				},
+			},
+			validateService: func(svc *corev1.Service) {
+				o.Expect(svc.Spec.Type).To(o.Equal(corev1.ServiceTypeClusterIP))
+				parsedClusterIP := net.ParseIP(svc.Spec.ClusterIP)
+				o.Expect(parsedClusterIP).ToNot(o.BeNil())
+			},
+			validateScyllaConfig: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				broadcastAddress, err := configClient.BroadcastAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
```

Comment:
nit: - `\n` (the next paragraph uses that pattern)

Reply:
done